test/e2e/knative_serving_test.go

```diff
@@ -73,6 +73,25 @@ func TestKnativeServing(t *testing.T) {
 		}
 	})
 
+	t.Run("verify kn ConcoleCLIDownload CO and deployment resources", func(t *testing.T) {
+		// Check the status of Deployment for kn ConsoleCLIDownload
+		if err := test.CheckDeploymentScale(caCtx, knativeServing, "kn-cli-downloads", "1"); err != nil {
+			t.Fatalf("failed to verify kn ConcoleCLIDownload deployment: %v", err)
+		}
+		// Verify that Route for kn ConsoleCLIDownload is ready and has a host
+		host, err := test.RouteHost(caCtx, knativeServing, "kn-cli-downloads")
+		if err != nil {
+			t.Fatalf("failed to verify kn ConcoleCLIDownload route: %v", err)
+		}
+		if host == "" {
+			t.Fatalf("kn ConsoleCLIDownload resource route is empty")
+		}
+		// Verify kn ConsoleCLIDownload CO and if download links are cluster local
+		if err := test.VerifyConcoleCLIDownload(caCtx, "kn", host); err != nil {
+			t.Fatalf("failed to verify kn ConsoleCLIDownload: %v", err)
+		}
```

Comment:
As a last test, would it be possible to:
1. Call the route and check the HTML contains the links the CCD contains?
2. Try the links themselves and see if they'd result in a download that has more than... dunno, 1MB? How big is kn? 😂 . HEAD requests might work for that.

I'd be fine adding these tests in subsequent PRs.

Reply:
Added checks to verify length of the links, if it starts with the route URL and downloads each to verify > 10MB.

Reply:
updated with to check content-length with HEAD requests, thanks for the suggestion and explaining me :+1: